### PR TITLE
chore(hardhat): refactor deploy task

### PR DESCRIPTION
**Pull request by henryacaron**

### CODE REVIEW

File: tasks/arena-deploy.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  const changes = new DiamondChanges();
+
+  const Verifier = (await deployFacet('Verifier', {}, hre)).address;
+  const LibGameUtils = (await deployFacet('LibGameUtils', {}, hre)).address;
+  const LibLazyUpdate = (await deployFacet('LibLazyUpdate', {}, hre)).address;
+  const LibArtifactUtils = (await deployFacet('LibArtifactUtils', { LibGameUtils }, hre)).address;
+  const LibPlanet = (await deployFacet('LibPlanet', { LibGameUtils, LibLazyUpdate, Verifier }, hre))
+    .address;
+
+  // const { LibGameUtils, LibArtifactUtils, LibPlanet } = await deployLibraries({}, hre);
```
Comment (cha0sg0d) [inline tasks/arena-deploy.ts:127]: why is library deploy commented out?

File: tasks/arena-deploy.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
+import { task, types } from 'hardhat/config';
```
Comment (cha0sg0d) [inline tasks/arena-deploy.ts:1]: @henryacaron Is this file a straight copy of `deploy.ts`?

File: tasks/utils.ts [hunk 3]
```diff
@@ -1,5 +1,9 @@
 import { subtask } from 'hardhat/config';
-import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { HardhatRuntimeEnvironment, Libraries } from 'hardhat/types';
+import * as path from 'path';
+import * as prettier from 'prettier';
+import * as fs from 'fs';
+import { tscompile } from '../utils/tscompile';
```
Comment (cha0sg0d) [inline tasks/utils.ts:7]: @henryacaron why are we adding all of these deploy tasks to utils? Can we put them in a new file for cleanliness sake?

File: tasks/utils.ts [hunk 4]
```diff
@@ -12,3 +16,141 @@ async function assertChainId({}, hre: HardhatRuntimeEnvironment) {
...
+   * This package contains deployed contract addresses, ABIs, and Typechain types
+   * for the Dark Forest game.
+   *
+   * ## Installation
+   *
+   * You can install this package using [\`npm\`](https://www.npmjs.com) or
+   * [\`yarn\`](https://classic.yarnpkg.com/lang/en/) by running:
+   *
+   * \`\`\`bash
+   * npm install --save @darkforest_eth/contracts
```
Comment (cha0sg0d) [inline tasks/utils.ts:75]:
```suggestion
   * npm install --save @dfdao/contracts
```
Will make README correct for installation of contracts

File: tasks/utils.ts [hunk 5]
```diff
@@ -12,3 +16,141 @@ async function assertChainId({}, hre: HardhatRuntimeEnvironment) {
...
+   * \`\`\`
+   * \`\`\`bash
+   * yarn add @darkforest_eth/contracts
```
Comment (cha0sg0d) [inline tasks/utils.ts:78]:
```suggestion
   * yarn add @dfdao/contracts
```
Will make README correct for installation of contracts

**Review by cha0sg0d — CHANGES_REQUESTED** [review]
hardhat config looks good. Confused about what `arena-deploy` is supposed to doing and think we should consider the best place to put deploy tasks currently in `utils.ts`.

File: tasks/arena-deploy.ts (hunk 2 above)
Comment (henryacaron) [reply tasks/arena-deploy.ts:1]: yes except the deployFacet lines are all different and we agreed not to overwrite any code

File: tasks/utils.ts (hunk 3 above)
Comment (henryacaron) [reply tasks/utils.ts:7]: they are deploy utilities so i figured they belonged best in utils

File: tasks/arena-deploy.ts (hunk 1 above)
Comment (henryacaron) [reply tasks/arena-deploy.ts:127]: it can be removed

Comment (cha0sg0d) [inline tasks/arena-deploy.ts:127]:
```suggestion
```

**Pull request closed without merging**